packages/hoppscotch-common/src/services/spotlight/searchers/environment.searcher.ts

```diff
@@ -370,10 +440,23 @@ export class SwitchEnvSpotlightSearcherService
   }
 
   onResultSelect(result: SpotlightSearcherResult): void {
-    const selectedEnvIndex = Number(result.id.split("-")[1])
-    setSelectedEnvironmentIndex({
-      type: "MY_ENV",
-      index: selectedEnvIndex,
-    })
+    if (result.id.startsWith("team-environment")) {
+      const teamAndEnvID = result.id.split("-")[2]
+      const [teamID, teamEnvID] = teamAndEnvID.split(":")
```

Comment:
We can maybe chain these. Feel free to keep it as is if you feel it affects readability.